pandas/core/arrays/arrow/array.py

```diff
@@ -451,6 +452,9 @@ def _cmp_method(self, other, op):
             result = pc_func(self._pa_array, other._pa_array)
         elif isinstance(other, (np.ndarray, list)):
             result = pc_func(self._pa_array, other)
+        elif isinstance(other, BaseMaskedArray):
```

Comment:
Hmm might this unintentionally make `ArrowExtentionArray` be prioritized over a custom 3rd party array?

Reply:
1000-1999 are for EAs.  I guess we could document that internal EAs only use up through e.g. 1499

Reply:
My understanding of `__pandas_priority__` was that it was specifically for non-internal EAs since at least for the internal ones we can dictate the return. Or did you imagine we are supposed to be using this internally as well?

Reply:
internally it doesn't really matter which method we used.  i was imagining we could use it internally for intra-EA ops but that wasn't a big selling point

Reply:
If this check and using `__pandas_priority__` are essentially similar I would slightly prefer to use this check to avoid spilling into interacting with 3rd party EA.__pandas_priority__ unless you feel strongly otherwise

Reply:
No objection